src/goto-programs/goto_program.cpp

```diff
@@ -668,6 +668,65 @@ bool goto_programt::instructiont::equals(const instructiont &other) const
   // clang-format on
 }
 
+bool goto_programt::instructiont::check_internal_invariants(
+  const symbol_tablet &table,
+  messaget &msg) const
+{
+  bool found_violation = false;
+  std::vector<std::string> id_collector;
+  auto symbol_finder = [&](const exprt &e) {
+    if(e.id() == ID_symbol)
+    {
+      auto symbol_expr = to_symbol_expr(e);
+      if(!table.has_symbol(symbol_expr.get_identifier()))
+        id_collector.push_back(id2string(symbol_expr.get_identifier()));
+    }
+  };
+
+  if(!table.has_symbol(function))
+    id_collector.push_back(id2string(function));
+
+  switch(type)
+  {
+  case GOTO:
```

Comment:
I think this whole construct can be simplified, unless there's a reason not to? just keep the `case ASSERT` and `case FUNCTION_CALL` that require special handling, and just remove the other ones and add one `default`?

Reply:
There's method in this madness. When we add further checks we simply put them under the right case, without having to add more cases. (At least I think so: this part I've also copied from #918.)

Reply:
Another thing to keep in mind is the semantics of what you've written. The behaviour is the same for all of `case GOTO`, `case ASSUME` and `case ASSERT` for example, where for each one of them `guard.visit(symbol_finder);` will run because of fall through of the switch cases. Is that intended, or only `case ASSERT` should demonstrate that? If yes that's okay. If no, then what I suggested is a better/cleaner idea, and semantically correct while reducing the possibility of having cases being unintendedly massed together.